drivers/usb/udc/udc_ambiq.c

```diff
@@ -25,6 +25,8 @@ enum udc_ambiq_event_type {
 	UDC_AMBIQ_EVT_HAL_IN_CMP,
 	/* Xfer request received via udc_ambiq_ep_enqueue API */
 	UDC_AMBIQ_EVT_XFER,
+	/* ZLP Pending to be transmitted */
+	UDC_AMBIQ_EVT_ZLP_PENDING,
```

Comment:
This commit should be squashed with the one that introduces the driver.
That you need a separate type for ZLP is suspicious.

Reply:
Hi @jfischer-no, Thanks for pointing this out. I was juggling on whether I should create a new event to trigger ZLP or just trigger the ZLP directly in complete event handler for IN transaction. 
Since you mentioned this as well, I think it's better to do the direct invocation instead. 
I have reverted to the said solution where there is no need for a separate `ZLP_PENDING` event